src/JsonApi/EventListener/TransformFieldsetsParametersListener.php

```diff
@@ -35,23 +36,47 @@ public function onKernelRequest(GetResponseEvent $event)
     {
         $request = $event->getRequest();
 
+        $includeParameter = $request->query->get('include');
         if (
             'jsonapi' !== $request->getRequestFormat() ||
             !($resourceClass = $request->attributes->get('_api_resource_class')) ||
-            !($fieldsParameter = $request->query->get('fields')) ||
-            !\is_array($fieldsParameter)
+            (!($fieldsParameter = $request->query->get('fields')) && !$includeParameter)
+        ) {
+            return;
+        }
+
+        if ($includeParameter && $request->attributes->get('_api_collection_operation_name')) {
```

Comment:
Can you explain why this exception is thrown here?
I think you should be able to include related resources for collections as seen in the example here - http://jsonapi.org/

Reply:
All example that I saw in the spec were using item only.
I'll update soon

Reply:
@Giant-Jelly It should now work also for collection, please give it a try